app/controllers/users/phone_setup_controller.rb

```diff
@@ -21,15 +21,20 @@ def create
       result = @user_phone_form.submit(user_phone_form_params)
       analytics.track_event(Analytics::MULTI_FACTOR_AUTH_PHONE_SETUP, result.to_h)
 
-      if result.success?
-        prompt_to_confirm_phone(id: nil, phone: @user_phone_form.phone)
-      else
-        render :index
-      end
+      result.success? ? handle_create_success(@user_phone_form.phone) : (render :index)
```

Comment:
Why change to the ternary here? It does not look like the broken out conditional should have given you any issues since you've reduced the complexity of it w/ `#handle_create_success`.

Reply:
It was one of the changes i made initially because the method was too long. This seems to me just as clear and more succinct.

Reply:
It is probably okay. I try to avoid ternary operators when the result of the conditional extends beyond returning a single variable (so not when invoking a function). Breaking up the logic over several lines helps me reason about each codepath on it's own. In the case of testing a form result, using a conditional also matches the pattern we see in all of our other controllers.

Reply:
changed back